### PR TITLE
Initialize function pointers at first use (II)

### PR DESCRIPTION
**Pull request by Dani-Hub**
Fixes static initialization issue of new compare function using interlocked functions. Kudus to Daniela Engert for her initial pull request ff8c321a655490783c7aac589796a5ee35a749e1.

### CODE REVIEW

File: src/operations.cpp [hunk 1]
```diff
@@ -614,8 +613,51 @@ namespace
...
+    return rtl_equal_unicode_string_api ? equal_string_ordinal_ic_1
+                                        : equal_string_ordinal_ic_2;
+  }
+
+  static Ptr_equal_string_ordinal_ic compare_function;
+  static long compare_function_state;
+
+  inline Ptr_equal_string_ordinal_ic compare()
+  {
+    for (unsigned i = 0; compare_function_state < 2; ++i)
```
Comment (Lastique) [inline src/operations.cpp]: This reads `compare_function_state` non-atomically (i.e. not thread-safe).

Comment (Dani-Hub) [reply src/operations.cpp]: Ouch, you are completely right! I thought that I could use the `<atomic>`-free interlocked approach that `boost/interprocess/detail/win32_api.hpp` uses, but now I realize that the corresponding code in the `function_address_holder::get_module` and `function_address_holder::get` is defect as well! I will take a look this night or tomorrow to improve the patch - thanks for the review so far!

Comment (Dani-Hub) [reply src/operations.cpp]: The non-atomical read access should have been fixed now via https://github.com/Dani-Hub/filesystem/commit/76c061ec79328cc4fee6a23f07e2918c66fe44a7

Comment (Dani-Hub) [reply src/operations.cpp]: I have now created ticket https://svn.boost.org/trac10/ticket/13309 to handle the same problem in `boost/interprocess/detail/win32_api.hpp` (twice)

**Pull request closed without merging**